File: src/_cpp/TowerDefenseBase/resources.cpp
//
//  IslandDefense
//
//  Created by Vladimir Tolmachev on 27.09.14.
//
//
#include "resources.h"


//const std::string kPlistGarage( "plist/garage.plist::" );

std::string kFont( "fonts/whitefont1.fnt" );
std::string kFontStroke( "fonts/mdefensefont2.fnt" );
std::string kFontHelvetica( "fonts/helvetica.fnt" );
std::string kFontHelveticaStroke( "fonts/helveticastoke.fnt" );


#ifdef _DEBUG 
#if CC_TARGET_PLATFORM == CC_PLATFORM_WIN32
const std::string kDirectoryToMaps( "ini/maps/map" );
#else
const std::string kDirectoryToMaps( "ini/maps/map" );
#endif
#else
const std::string kDirectoryToMaps( "ini/maps/map" );
#endif

//
//		Sounds
//
#if CC_TARGET_PLATFORM == CC_PLATFORM_IOS
const std::string kPathSound( "audio/sound/" );
const std::string kPathMusic( "audio/music/" );
const std::string kSoundsEXT( ".m4a" );
const std::string kMusicEXT( ".m4a" );
#endif
#if CC_TARGET_PLATFORM == CC_PLATFORM_ANDROID
const std::string kPathSound( "audio/sound_ogg/" );
const std::string kPathMusic( "audio/music_ogg/" );
const std::string kSoundsEXT( ".ogg" );
const std::string kMusicEXT( ".ogg" );
#endif
#if CC_TARGET_PLATFORM == CC_PLATFORM_WIN32
const std::string kPathSound( "../Resources.win/audio/sound/" );
const std::string kPathMusic( "../Resources.win/audio/music/" );
const std::string kSoundsEXT( ".wav" );
const std::string kMusicEXT( ".wav" );
#endif
#if CC_TARGET_PLATFORM == CC_PLATFORM_MAC
const std::string kPathSound( "audio/sound/" );
const std::string kPathMusic( "audio/music/" );
const std::string kSoundsEXT( ".m4a" );
const std::string kMusicEXT( ".wav" );
#endif


//const std::string kMusicInterface = kPathMusic + "track0" + kMusicEXT;
//const std::string kMusicGame0 = kPathMusic + "game0" + kMusicEXT;
const std::string kMusicMainMenu( "##music_mainmenu##" );
const std::string kMusicMap( "##music_map##" );
const std::string kMusicGamePeace( "##music_gamepeace##" );
const std::string kMusicGameBattle( "##music_gamebattle##" );
const std::string kMusicVictory( "##music_victory##" );
const std::string kMusicDefeat( "##music_defeat##" );

const std::string kSoundGameTowerBuy( kPathSound + "Building" + kSoundsEXT );
const std::string kSoundGameTowerUpgrade( kPathSound + "Building" + kSoundsEXT );
const std::string kSoundGameTowerPlaceActivate( kPathSound + "Clearing" + kSoundsEXT );
const std::string kSoundGameTowerPlaceSelect( kPathSound + "Click_2" + kSoundsEXT );
const std::string kSoundGameWaveIcon( kPathSound + "Click" + kSoundsEXT );
const std::string kSoundShopShow( kPathSound + "Shop_Open" + kSoundsEXT );
const std::string kSoundShopHide( kPathSound + "Shop_Close" + kSoundsEXT );
const std::string kSoundShopPurchase( kPathSound + "Buying" + kSoundsEXT );
const std::string kSoundLabUpgrade( kPathSound + "Upgrade_01" + kSoundsEXT );

const std::string kSoundGamePauseOn( kPathSound + "game_pause_on" + kSoundsEXT );
const std::string kSoundGamePauseOff( kPathSound + "game_pause_off" + kSoundsEXT );
const std::string kSoundGameFastModeOn( kPathSound + "game_fastmode_on" + kSoundsEXT );
const std::string kSoundGameFastModeOff( kPathSound + "game_fastmode_off" + kSoundsEXT );
const std::string kSoundGameTowerSelect( kPathSound + "game_select_tower" + kSoundsEXT );
const std::string kSoundGameTowerBuyCancel( kPathSound + "" + kSoundsEXT );
const std::string kSoundGameTowerSelling( kPathSound + "game_tower_selling" + kSoundsEXT );
const std::string kSoundGameTowerUpgradeCancel( kPathSound + "game_tower_upgrade_cancel" + kSoundsEXT );
const std::string kSoundGameMoneyAdd( kPathSound + "game_money_add" + kSoundsEXT );
const std::string kSoundGameCrystalAdd( kPathSound + "game_crystal_add" + kSoundsEXT );
const std::string kSoundGameFinishSuccess( kPathSound + "game_win" + kSoundsEXT );
const std::string kSoundGameFinishFailed( kPathSound + "game_fail" + kSoundsEXT );
const std::string kSoundStatisticAppearance( kPathSound + "statistic_appearance" + kSoundsEXT );
const std::string kSoundAppearanceText( kPathSound + "appearance_text" + kSoundsEXT );
const std::string kSoundAppearanceImage( kPathSound + "appearance_image" + kSoundsEXT );
const std::string kSoundGameStart( kPathSound + "game_start" + kSoundsEXT );
const std::string kSoundGarageUpgradeYes( kPathSound + "garage_upgrade_yes" + kSoundsEXT );
const std::string kSoundGarageUpgradeCancel( kPathSound + "garage_upgrade_cancel" + kSoundsEXT );

const std::string kSoundTowerAckAckShoot( kPathSound + "ackack_shoot" + kSoundsEXT );
const std::string kSoundTowerAckAckBulletHit( kPathSound + "ackack_bullet_hit" + kSoundsEXT );
const std::string kSoundTowerFlaregunShoot( kPathSound + "flaregun_shoot" + kSoundsEXT );
const std::string kSoundTowerFlaregunRacketHit( kPathSound + "flaregun_racket_hit" + kSoundsEXT );
const std::string kSoundTowerFlaregunCocking( kPathSound + "flaregun_cocking" + kSoundsEXT );
const std::string kSoundTowerFlaregunRelax( kPathSound + "flaregun_relax" + kSoundsEXT );
const std::string kSoundTowerFiregunShoot( kPathSound + "firegun_shoot" + kSoundsEXT );
const std::string kSoundTowerTeslaShoot( kPathSound + "tesla_shoot" + kSoundsEXT );
const std::string kSoundTowerTeslaRelax( kPathSound + "tesla_relax" + kSoundsEXT );
const std::string kSoundTowerLaserShoot( kPathSound + "laser_shoot" + kSoundsEXT );
const std::string kSoundTowerLaserCharge( kPathSound + "laser_charge" + kSoundsEXT );
const std::string kSoundTowerLaserCocking( kPathSound + "laser_cocking" + kSoundsEXT );
const std::string kSoundTowerLaserRelax( kPathSound + "laser_relax" + kSoundsEXT );
const std::string kSoundTowerNitrogenShoot( kPathSound + "nitrogen_shoot" + kSoundsEXT );
const std::string kSoundTowerHowitzerShoot( kPathSound + "howitzer_shoot" + kSoundsEXT );

File: src/_eu/resources.js
//Define namespace
var EU = EU || {};

EU.kPathSpriteSquare = "images/square.png";
EU.kFont =  "fonts/whitefont1.fnt" ;
EU.kFontStroke =  "fonts/mdefensefont2.fnt" ;
EU.kFontHelvetica =  "fonts/helvetica.fnt" ;
EU.kFontHelveticaStroke =  "fonts/helveticastoke.fnt" ;

if(EU._DEBUG) {
    if (EU.CC_TARGET_PLATFORM == EU.CC_PLATFORM_WIN32)
        EU.kDirectoryToMaps =  "ini/maps/map" ;
    else
        EU.kDirectoryToMaps =  "ini/maps/map" ;
}
else {
    EU.kDirectoryToMaps =  "ini/maps/map" ;
}

//
//		Sounds
//
if (EU.CC_TARGET_PLATFORM == EU.CC_PLATFORM_IOS){
    EU.kPathSound =  "audio/sound/" ;
    EU.kPathMusic =  "audio/music/" ;
    EU.kSoundsEXT =  ".m4a" ;
    EU.kMusicEXT =  ".m4a" ;
}
if (EU.CC_TARGET_PLATFORM == EU.CC_PLATFORM_ANDROID){

    EU.kPathSound =  "audio/sound_ogg/" ;
    EU.kPathMusic =  "audio/music_ogg/" ;
    EU.kSoundsEXT =  ".ogg" ;
    EU.kMusicEXT =  ".ogg" ;
}
if (EU.CC_TARGET_PLATFORM == EU.CC_PLATFORM_WIN32){

    EU.kPathSound =  "../Resources.win/audio/sound/" ;
    EU.kPathMusic =  "../Resources.win/audio/music/" ;
    EU.kSoundsEXT =  ".wav" ;
    EU.kMusicEXT =  ".wav" ;
}
if (EU.CC_TARGET_PLATFORM == EU.CC_PLATFORM_MAC){
    EU.kPathSound =  "audio/sound/" ;
    EU.kPathMusic =  "audio/music/" ;
    EU.kSoundsEXT =  ".m4a" ;
    EU.kMusicEXT =  ".wav" ;
}


//const std::string EU.kMusicInterface = EU.kPathMusic + "track0" + EU.kMusicEXT;
//const std::string EU.kMusicGame0 = EU.kPathMusic + "game0" + EU.kMusicEXT;
EU.kMusicMainMenu =  "##music_mainmenu##" ;
EU.kMusicMap =  "##music_map##" ;
EU.kMusicGamePeace =  "##music_gamepeace##" ;
EU.kMusicGameBattle =  "##music_gamebattle##" ;
EU.kMusicVictory =  "##music_victory##" ;
EU.kMusicDefeat =  "##music_defeat##" ;

EU.kSoundGameTowerBuy =  EU.kPathSound + "Building" + EU.kSoundsEXT ;
EU.kSoundGameTowerUpgrade =  EU.kPathSound + "Building" + EU.kSoundsEXT ;
EU.kSoundGameTowerPlaceActivate =  EU.kPathSound + "Clearing" + EU.kSoundsEXT ;
EU.kSoundGameTowerPlaceSelect =  EU.kPathSound + "Click_2" + EU.kSoundsEXT ;
EU.kSoundGameWaveIcon =  EU.kPathSound + "Click" + EU.kSoundsEXT ;
EU.kSoundShopShow =  EU.kPathSound + "Shop_Open" + EU.kSoundsEXT ;
EU.kSoundShopHide =  EU.kPathSound + "Shop_Close" + EU.kSoundsEXT ;
EU.kSoundShopPurchase =  EU.kPathSound + "Buying" + EU.kSoundsEXT ;
EU.kSoundLabUpgrade =  EU.kPathSound + "Upgrade_01" + EU.kSoundsEXT ;

EU.kSoundGamePauseOn =  EU.kPathSound + "game_pause_on" + EU.kSoundsEXT ;
EU.kSoundGamePauseOff =  EU.kPathSound + "game_pause_off" + EU.kSoundsEXT ;
EU.kSoundGameFastModeOn =  EU.kPathSound + "game_fastmode_on" + EU.kSoundsEXT ;
EU.kSoundGameFastModeOff =  EU.kPathSound + "game_fastmode_off" + EU.kSoundsEXT ;
EU.kSoundGameTowerSelect =  EU.kPathSound + "game_select_tower" + EU.kSoundsEXT ;
EU.kSoundGameTowerBuyCancel =  EU.kPathSound + "" + EU.kSoundsEXT ;
EU.kSoundGameTowerSelling =  EU.kPathSound + "game_tower_selling" + EU.kSoundsEXT ;
EU.kSoundGameTowerUpgradeCancel =  EU.kPathSound + "game_tower_upgrade_cancel" + EU.kSoundsEXT ;
EU.kSoundGameMoneyAdd =  EU.kPathSound + "game_money_add" + EU.kSoundsEXT ;
EU.kSoundGameCrystalAdd =  EU.kPathSound + "game_crystal_add" + EU.kSoundsEXT ;
EU.kSoundGameFinishSuccess =  EU.kPathSound + "game_win" + EU.kSoundsEXT ;
EU.kSoundGameFinishFailed =  EU.kPathSound + "game_fail" + EU.kSoundsEXT ;
EU.kSoundStatisticAppearance =  EU.kPathSound + "statistic_appearance" + EU.kSoundsEXT ;
EU.kSoundAppearanceText =  EU.kPathSound + "appearance_text" + EU.kSoundsEXT ;
EU.kSoundAppearanceImage =  EU.kPathSound + "appearance_image" + EU.kSoundsEXT ;
EU.kSoundGameStart =  EU.kPathSound + "game_start" + EU.kSoundsEXT ;
EU.kSoundGarageUpgradeYes =  EU.kPathSound + "garage_upgrade_yes" + EU.kSoundsEXT ;
EU.kSoundGarageUpgradeCancel =  EU.kPathSound + "garage_upgrade_cancel" + EU.kSoundsEXT ;

EU.kSoundTowerAckAckShoot =  EU.kPathSound + "ackack_shoot" + EU.kSoundsEXT ;
EU.kSoundTowerAckAckBulletHit =  EU.kPathSound + "ackack_bullet_hit" + EU.kSoundsEXT ;
EU.kSoundTowerFlaregunShoot =  EU.kPathSound + "flaregun_shoot" + EU.kSoundsEXT ;
EU.kSoundTowerFlaregunRacketHit =  EU.kPathSound + "flaregun_racket_hit" + EU.kSoundsEXT ;
EU.kSoundTowerFlaregunCocking =  EU.kPathSound + "flaregun_cocking" + EU.kSoundsEXT ;
EU.kSoundTowerFlaregunRelax =  EU.kPathSound + "flaregun_relax" + EU.kSoundsEXT ;
EU.kSoundTowerFiregunShoot =  EU.kPathSound + "firegun_shoot" + EU.kSoundsEXT ;
EU.kSoundTowerTeslaShoot =  EU.kPathSound + "tesla_shoot" + EU.kSoundsEXT ;
EU.kSoundTowerTeslaRelax =  EU.kPathSound + "tesla_relax" + EU.kSoundsEXT ;
EU.kSoundTowerLaserShoot =  EU.kPathSound + "laser_shoot" + EU.kSoundsEXT ;
EU.kSoundTowerLaserCharge =  EU.kPathSound + "laser_charge" + EU.kSoundsEXT ;
EU.kSoundTowerLaserCocking =  EU.kPathSound + "laser_cocking" + EU.kSoundsEXT ;
EU.kSoundTowerLaserRelax =  EU.kPathSound + "laser_relax" + EU.kSoundsEXT ;
EU.kSoundTowerNitrogenShoot =  EU.kPathSound + "nitrogen_shoot" + EU.kSoundsEXT ;
EU.kSoundTowerHowitzerShoot =  EU.kPathSound + "howitzer_shoot" + EU.kSoundsEXT ;
